hal/src/stm32f2xx/sleep_hal.cpp

```diff
@@ -134,6 +149,18 @@ static int configRtcWakeupSource(const hal_wakeup_source_base_t* wakeupSources)
     return SYSTEM_ERROR_NONE;
 }
 
+static int configUsartWakeupSource(const hal_wakeup_source_base_t* wakeupSources, uint8_t* configured) {
+    auto source = wakeupSources;
+    while (source) {
+        if (source->type == HAL_WAKEUP_SOURCE_TYPE_USART) {
+            *configured = 1;
+            // We don't need to configure anything, as the RX interrupt is enabled when the USART is enabled.
```

Comment:
I'm pretty sure we need to do at least something here. Everything still gets clocked and other interrupts than USART are enabled. This is not a problem with `PWR_Regulator_LowPower`, because only RTC and EXTI are clocked, but with the normal WFI mode this is not going to work. We resolve this problem on Gen 3 with BASEPRI. Something similar probably needs to be done here.

Moreover, any other USART event may wake us up.